content/en/blog/_posts/qos-class-resources.md

```diff
@@ -0,0 +1,302 @@
+write bandwidth throttled.
+
+`/etc/crio/blockio.conf.yaml`:
+
+```yaml
+Classes:
+  high-prio:
+    - Weight: 500
+  normal:
+    - Devices:
+        - "/dev/sd[a-z]"
+      throttleWriteBps: "60M"
+  low-prio:
+    - Weight: 50
+    - Devices:
+        - "/dev/sd[a-z]"
+      throttleReadBps: "50M"
+      throttleWriteBps: "10M"
+```
+
+#### Starting CRI-O
+
+CRI-O needs to be configured to use the rdt and blockio configuration specified
+above which can be done with `rdt_config_file` and `blockio_config_file` config
+file options or with the `--rdt-config-file` and `-–blockio-config-file`
+command line options.
+
+`/etc/crio/crio.conf`:
+
+```toml
+...
+rdt_config_file = "/etc/crio/rdt.conf.yaml"
+
+blockio_config_file = "/etc/crio/blockio.conf.yaml"
+...
+```
+
+#### Assigning QoS-class resources
+
+Below is a referential example leveraging both RDT and blockio resources. It
+creates a Pod of three containers with:
+
+- all containers are assigned to the “limited” RDT class by default, except for
+  the “data-path” container getting assigned to “unlimited” class allowing full
+  cache usage
+- all containers are assigned to the “normal” blockio class by default, except
+  for the “logger” container getting assigned to the “low-prio” class
+  throttling its disk usage more strictly
+
+`demo-pod.yaml:`
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: demo
+  annotations:
+    rdt.resources.beta.kubernetes.io/pod: limited
+    rdt.resources.beta.kubernetes.io/container.data-path: unlimited
+    blockio.resources.beta.kubernetes.io/pod: normal
+    blockio.resources.beta.kubernetes.io/container.looger: low-prio
+spec:
+  containers:
+  - name: data-path
+    image: k8s.gcr.io/pause
+  - name: control-path
+    image: k8s.gcr.io/pause
+  - name: logger
+    image: k8s.gcr.io/pause
+```
+
+```bash
+kubectl apply -f demo-pod.yaml
+```
+
+Note that the containers are minimal pause images, doing nothing in practice,
+but just demonstrating how different containers are assigned to different
+classes.
+
+### Limitations of the runtime-only approach
+
+The biggest limitations in this runtime-only approach probably lies in the user
+interface. Users have no visibility into what QoS-class resource types (or
+classes within) are available on the nodes and there is no support in
+Kubernetes scheduler.
+
+## How to participate
+
+The enhancement proposal includes more details of the feature and the planned
+implementation.
+
+Please join SIG Node to get involved in the design and development. You can
+participate for example by commenting the [KEP][kep] directly, in
+[#sig-node][sig-node-slack] slack or replying to the
+[discussion thread in google groups][google-groups-thread]. We’d be delighted
+to hear more opinions and ideas how and where QoS-class resources could be used
+and how the support should be implemented.
+
+<!-- References -->
+[kep]: https://github.com/kubernetes/enhancements/pull/3004
```

Comment:
```suggestion
[kep]: https://github.com/kubernetes/enhancements/issues/3008
```
The KEP will be KEP 3008 (not KEP 3004).

Reply:
Thanks, fixed